src/setters/price.ts

```diff
@@ -52,6 +53,20 @@ export const setGvtPrice = (): void => {
     price.save();
 }
 
+export const set3CrvPrice = (): void => {
+    let crvPrice = NUM.ZERO;
+    const contract = ThreePool.bind(ADDR.THREE_POOL);
+    const virtualPrice = contract.try_get_virtual_price();
+    if (virtualPrice.reverted) {
+        log.error('Get virtual price for 3crv failed', []);
+    }
```

Comment:
You need an **else** here; otherwise, if the virtualPrice reverts, you will try to assign a value that is empty and the subgraph will fail and stop syncing.
On the other hand, you don't need to initialise crvPrice to zero at the beginning.
In summary:
```
export const set3CrvPrice = (): void => {
    const contract = ThreePool.bind(ADDR.THREE_POOL);
    const virtualPrice = contract.try_get_virtual_price();
    if (virtualPrice.reverted) {
        log.error('Get virtual price for 3crv failed', []);
    } else {
       const crvPrice = tokenToDecimal(virtualPrice.value, 18, 7);
       let price = initPrice();
       price.threeCrv = crvPrice;
       price.save();
    }
}

Reply:
Your suggestion the **else** is better, because the 3crv's price is initialize with zero not need assign zero again. I didn't understand why the subgraph will fail and stop syncing if not the **else**?